halpybot/packages/utils/spansh.py

```diff
@@ -0,0 +1,150 @@
+
+    # Encode spaces to be URL compatible
+    url_a = quote(sanitize_system_name(points.point_a.name))
+    url_b = quote(sanitize_system_name(points.point_b.name))
+    # Format spansh results URL with parameters
+    short = f"{config.spansh.page_endpoint}/{jobs[1]}?efficiency=60&from={url_a}&to={url_b}&range={points.jump_range}"
+    if config.yourls.enabled:
+        short = await shorten(short)  # Shorten the URL if the yourls module is enabled
+    return await ctx.reply(f"Here's a spansh URL: {short}")
+
+
+async def spansh(ctx: Context, points: Points) -> None:
+    """
+    Starts calculating the normal and Neutron Jump Count using spansh.co.uk
+
+    Args:
+        ctx (Context): PYDLE Context
+        points (Points): A pair of EDSM valid point locations and names, with a jump range
+
+    Returns:
+        None
+
+    Raises:
+        SpanshNoResponse: spansh did not respond in time.
+        SpanshBadResponse: Spansh returned an unprocessable response
+    """
+    efficiency = [100, 60]
+    job_id = []
+    for percent in efficiency:
+        # Create request parameters for both Normal and Neutron Jump Calculations
+        params = {
+            "efficiency": percent,
+            "range": points.jump_range,
+            "from": sanitize_system_name(points.point_a.name),
+            "to": sanitize_system_name(points.point_b.name),
+        }
+        try:  # Try to start Jump Calculations
+            responses = await web_get(config.spansh.route_endpoint, params)
+        except aiohttp.ClientError as ex:
+            logger.exception(
+                f"spansh did not respond while trying to start the {percent}% efficiency jump count calculation"
+            )
+            raise SpanshNoResponse from ex
+        if "error" in responses:  # Process errors received from spansh
+            logger.warning(f"Spansh encountered an error: {responses['error']}")
+            if responses["error"] == "Could not find starting system":
+                return await ctx.reply("Spansh was unable to find the Starting System.")
+            if responses["error"] == "Could not find finishing system":
+                return await ctx.reply("Spansh was unable to find the Target System.")
+            # Other possible errors are issues with Range or efficiency,
+            # those should not be possible as Range has been preprocessed and efficiencies are constants
+            return await ctx.reply(
+                "Spansh encountered an error processing and was unable to continue."
+            )
+        if "job" not in responses:
+            # Spansh returned neither a job nor an error, something is wrong
+            raise SpanshBadResponse
+        job_id.append(responses["job"])
+    # Calculations have been started, start checking and processing results
+    asyncio.create_task(spansh_get_routes(ctx, points, job_id))
+    return await ctx.reply("Spansh calculations have been started...")
```

Comment:
possible task exception is never retrieved.

Reply:
I think we're going to tolerate this one here.